src/components/Search.js

```diff
@@ -18,8 +19,26 @@ class Search extends React.Component {
         super(props);
         this.searchForm = React.createRef();
         this.searchInput = React.createRef();
+        this.state = {
+            autocompleteEvent: '',
+            autocompleteValue: ''
+        };
+    }
+
+    logEvent() {
+        const {autocompleteEvent, autocompleteValue} = this.state;
+        console.log(`Event: ${autocompleteEvent} : ${autocompleteValue}`);
+    }
+
+    handleOnSelect(val) {
+        this.setState({autocompleteEvent: 'onSelect'});
+        this.setState({autocompleteValue: val}, () => { this.logEvent() });
```

Comment:
These 2 setStates should be combined:

`this.setState({autocompleteEvent: 'onSelect', autocompleteValue: val}})`